src/Functions/ignore.cpp

```diff
@@ -49,11 +49,6 @@ class FunctionIgnore : public IFunction
     {
         return DataTypeUInt8().createColumnConst(input_rows_count, 0u);
     }
-
-    ColumnPtr getResultIfAlwaysReturnsConstantAndHasArguments(const ColumnsWithTypeAndName &) const override
```

Comment:
@KochetovNicolai  this method was never get called, so ActionsDAG Node allow_constant_folding was always true.

Reply:
How about here?
https://github.com/ClickHouse/ClickHouse/blob/b0476c1fa219ca98f660d44f254eb4ed9e4fc12e/src/Interpreters/ActionsDAG.cpp#L232

Reply:
But I see it is not needed now.